presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -747,4 +749,67 @@ public boolean cancel(boolean mayInterruptIfRunning)
             return false;
         }
     }
+
+    // Helper class for combining rows from multiple pages into a single List.
+    // Not thread safe.
+    // TODO: Replace RuntimeExceptions with better alternatives.
+    private static class RowData
+    {
+        private List<List<Object>> rows = new ArrayList<>();
+        private boolean rowsFinalized;
+        private long dataSize;
```

Comment:
The class looks like a wrapper for `ImmutableList.Builder<List<Object>> rows` and `long dataSize`.
I think it would be more readable to use such 2 variables directly in the code. Builder has explicit lifecycle, here you manage lifecycle with `rowsFinalized` flag